src/main/java/net/glowstone/block/itemtype/ItemBow.java

```diff
@@ -103,8 +103,12 @@ public void endUse(GlowPlayer player, ItemStack bow) {
             if (ThreadLocalRandom.current().nextDouble() < 0.2) {
                 launchedArrow.setCritical(true);
             }
-            // TODO: Is main hand always the correct slot?
-            player.setItemInHand(InventoryUtil.damageItem(player, bow));
+
+            if (player.getItemInHand() == bow) {
```

Comment:
The way Vanilla functionality works, you can predict in which slot the bow was in:

- If the item type of the main slot is `Material.BOW`, the main slot was used.
- Else, the off slot was used.

(The main slot takes priority, so even if the player has bows in both hands, you can expect that the one in the main hand was used).

This avoids the risk that the bow item is changed in the meantime, thus potentially breaking any equality check.